In [1]:
import pandas as pd

df = pd.read_csv('insurance_claims_cleaned_fixed.csv', dtype={'column_name': 'string'})

We will be looking at hour of the day, insured occupation, and insured education level as they relate to fraud reported

#lets first get some info on insured hour of the day as it relates to fraud reported

In [2]:
#lets get fraud rate reported by hour of the day and put it into percentages and into a table

#now lets see the percentages in a table
pd.crosstab(df['incident_hour_of_the_day'], df['fraud_reported'], normalize = 'columns')


fraud_reported                  No       Yes
incident_hour_of_the_day                    
0                         0.053121  0.048583
1                         0.027888  0.032389
2                         0.027888  0.040486
3                         0.054449  0.048583
4                         0.050465  0.032389
5                         0.031873  0.036437
6                         0.039841  0.056680
7                         0.042497  0.032389
8                         0.039841  0.024291
9                         0.051793  0.016194
10                        0.041169  0.060729
11                        0.023904  0.048583
12                        0.042497  0.032389
13                        0.050465  0.032389
14                        0.037185  0.060729
15                        0.034529  0.052632
16                        0.043825  0.064777
17                        0.061089  0.032389
18                        0.035857  0.056680
19                        0.041169  0.036437
20                        0.033201  0.036437
21                        0.042497  0.040486
22                        0.043825  0.020243
23                        0.049137  0.056680

Fraud incidents are more concentrated in late morning to early evening hours (10:00–18:00).

Highest proportions occur at 10 (6.1%), 14 (6.1%), 16 (6.5%), 18 (5.7%), and 23 (5.7%).

Early morning (0–5 hrs) shows relatively lower fraud proportions, typically around 3–5% of total fraud cases.

Legitimate claims are more evenly spread throughout the day, with modest peaks around 0, 3, 9, and 17 hrs.

Midday (10–16 hrs) stands out as the most fraud-heavy window, suggesting possible behavioral or reporting pattern differences.

In [3]:
#now lets take a look at insured occupation against fraud reported


#lets do a crosstab to see the percentages
pd.crosstab(df['insured_occupation'], df['fraud_reported'], normalize = 'columns')

fraud_reported            No       Yes
insured_occupation                    
adm-clerical        0.071713  0.044534
armed-forces        0.069057  0.068826
craft-repair        0.069057  0.089069
exec-managerial     0.063745  0.113360
farming-fishing     0.049137  0.064777
handlers-cleaners   0.057105  0.044534
machine-op-inspct   0.094290  0.089069
other-service       0.078353  0.048583
priv-house-serv     0.078353  0.048583
prof-specialty      0.088977  0.072874
protective-serv     0.065073  0.056680
sales               0.073041  0.085020
tech-support        0.074369  0.089069
transport-moving    0.067729  0.085020

Executive/Managerial (11.3%) — nearly double their share of legitimate claims (6.4%), suggesting potential for strategic or opportunistic fraud among higher-income policyholders.

Professional Specialty (7.3%), Craft/Repair (8.9%), Tech Support (8.9%), and Sales (8.5%) also show above-average fraud rates.

These roles often involve knowledge of systems or insurance processes, possibly enabling more calculated claims.

Fraud is more prevalent among skilled or decision-making professions, while routine or service-oriented roles skew legitimate

In [4]:
#finally, lets take a look at insured education level against fraud reported

pd.crosstab(df['insured_education_level'], df['fraud_reported'], normalize = 'columns')

fraud_reported                 No       Yes
insured_education_level                    
Associate                0.147410  0.137652
College                  0.119522  0.129555
High School              0.164675  0.145749
JD                       0.158035  0.170040
MD                       0.140770  0.153846
Masters                  0.147410  0.129555
PhD                      0.122178  0.133603

Fraud proportions rise gradually from High School (14.6%) up to JD (17.0%) and MD (15.4%), suggesting a modest positive correlation between higher education and fraud likelihood.

These groups may possess greater awareness of policy terms or loopholes, enabling more sophisticated claim manipulation.

Fraudulent activity is evenly spread across education levels, but tends to increase slightly with advanced degrees — hinting that fraud may hinge more on opportunity and sophistication than on basic socioeconomic status